server/src/main/java/com/linecorp/centraldogma/server/storage/project/InternalProjectInitializer.java

```diff
@@ -30,28 +31,52 @@
 import com.linecorp.centraldogma.common.ChangeConflictException;
 import com.linecorp.centraldogma.common.Markup;
 import com.linecorp.centraldogma.common.ProjectExistsException;
+import com.linecorp.centraldogma.common.ReadOnlyException;
 import com.linecorp.centraldogma.common.RepositoryExistsException;
 import com.linecorp.centraldogma.common.Revision;
 import com.linecorp.centraldogma.internal.Jackson;
 import com.linecorp.centraldogma.server.command.CommandExecutor;
 import com.linecorp.centraldogma.server.metadata.MetadataService;
 import com.linecorp.centraldogma.server.metadata.Tokens;
-import com.linecorp.centraldogma.server.storage.project.Project;
 
-public final class ProjectInitializer {
+/**
+ * Initializes the internal project and repositories.
+ */
+public final class InternalProjectInitializer {
 
     public static final String INTERNAL_PROJECT_DOGMA = "dogma";
 
+    private final CommandExecutor executor;
+
+    /**
+     * Creates a new instance.
+     */
+    public InternalProjectInitializer(CommandExecutor executor) {
+        this.executor = executor;
+    }
+
     /**
      * Creates an internal project and repositories such as a token storage.
      */
-    public static void initializeInternalProject(CommandExecutor executor) {
+    public void initialize() {
         final long creationTimeMillis = System.currentTimeMillis();
-        initializeInternalProject(executor, creationTimeMillis, INTERNAL_PROJECT_DOGMA);
+        try {
+            executor.execute(createProject(creationTimeMillis, Author.SYSTEM, INTERNAL_PROJECT_DOGMA))
+                    .get();
+        } catch (Throwable cause) {
+            final Throwable peeled = Exceptions.peel(cause);
+            if (peeled instanceof ReadOnlyException) {
+                // The executor has stopped right after starting up.
+                return;
```

Comment:
Question to make sure I'm understanding correctly. Does this scenario occur because of the server status update? If a server doesn't have the necessary internal files but an exception isn't thrown here, could a CD server exist without the `dogma` project?

Reply:
> Does this scenario occur because of the server status update?

Right. `"server-status.properties"` in a server was `WRITABLE` but `READ_ONLY` was applied from reply logs.

> could a CD server exist without the dogma project?

No, impossible. However, a server's status mode can be changed after the server starts successfully. I don't think it will be a problem for new servers because the existing data in a replica should be copied to the new ones before starting them.